bsdlib/doc/at_commands.rst

```diff
@@ -0,0 +1,57 @@
+.. _at_commands:
+
+Sending AT commands
+###################
+
+You can use the BSD socket interface to send AT commands to the LTE modem and to receive the responses.
+
+To do so, create a socket with the proprietary address family NRF_AF_LTE and the protocol family NRF_PROTO_AT.
+You can then write AT command strings to that socket, and they are sent to the modem.
+The responses are available when you read from the socket.
+Received AT messages are not truncated; to read the full response, provide a sufficiently large buffer or fetch the full response in several read operations.
+
+See the `AT Commands Reference Guide <https://infocenter.nordicsemi.com/topic/ref_at_commands/REF/at_commands/intro.html>`_ for detailed information on the available AT commands.
+
+The following BSD socket functions are available for the NRF_PROTO_AT protocol family:
+
+- nrf_socket for creating a socket
+- nrf_close for closing a socket
+- nrf_fcntl for managing socket options
+- nrf_read for reading data from a socket
+- nrf_recv for reading data from a socket and concatenating several received messages into one receive buffer
+- nrf_write and nrf_send for writing data to a socket
+
+By default, read and write functions are blocking.
+To make them non-blocking, use nrf_fcntl to set the NRF_O_NONBLOCK flag, or pass NRF_MSG_DONTWAIT as flag to the function call.
+
+The following code example shows how to send an AT command and receive the response:
+
+.. code-block:: c
+
+   #include "nrf_socket.h" // socket(), NRF_AF_LTE family type, NRF_PROTO_AT protocol.
+   #include <string.h>     // strncmp()
+
+   int func(void)
+   {
+       // Create a socket for AT commands.
+       int fd = nrf_socket(NRF_AF_LTE, 0, NRF_PROTO_AT);
+
+       // Write the AT command.
+       nrf_write(fd, "AT+CEREG=2", 10);
+
+       // Allocate a response buffer.
+       char ok_buffer[10];
+
+       // Read an AT message. More than 2 bytes to consume an entire message.
```

Comment:
What does "More than 2 bytes to consume an entire message." mean?

Reply:
The assumtion here is that "OK" is returned from the modem, and to read OK you must read at least 2 bytes from the socket. So I think it's a way to explain the last parameter to the nrf_read function.